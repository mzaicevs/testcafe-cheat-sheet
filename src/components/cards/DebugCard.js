import React from "react";
import { debug } from "../../snippets";
import CodeCard from "../../base_components/CodeCard";

const DebugCard = () => (
  <CodeCard
    cardTitle={debug.title}
    fullDocLink={debug.fullDocLink}
    codeSnippet={{
      code: debug.code,
      language: debug.language,
      example: debug.example
    }}
    description={debug.description}
  />
);

export default DebugCard;
