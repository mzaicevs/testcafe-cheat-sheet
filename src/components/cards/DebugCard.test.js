import React from "react";
import { shallow } from "enzyme";
import DebugCard from "./DebugCard";

jest.mock("react-syntax-highlighter/dist/esm/styles/hljs", () => () => {});

describe("DebugCard", () => {
  it("matches snapshot", () => {
    const wrapper = shallow(<DebugCard />);
    expect(wrapper).toMatchSnapshot();
  });
});
