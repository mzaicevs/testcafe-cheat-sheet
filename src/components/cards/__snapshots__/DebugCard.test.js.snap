// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DebugCard matches snapshot 1`] = `
<CodeCard
  cardTitle="Debug"
  codeSnippet={
    Object {
      "code": Array [
        "t.debug()",
      ],
      "example": "fixture \`Debugger example\`
  .page \`http://devexpress.github.io/testcafe/example/\`;

test('Debugger', async t => {
  await t
      .debug()
      .click('#populate')
      .click('#submit-button');
});",
      "language": "javascript",
    }
  }
  description="Pauses the test and allows you to debug using the browser's developer tools."
  fullDocLink="https://devexpress.github.io/testcafe/documentation/test-api/debugging.html"
/>
`;
